int32 gamecase
float64 ball_dt
